simulator/core/perf_sim.cpp

```diff
@@ -161,6 +198,20 @@ void PerfSim<ISA>::clock_decode( Cycle cycle)
     /* receive flush signal */
     const bool is_flush = rp_decode_flush->is_ready( cycle) && rp_decode_flush->read( cycle);
 
+    /* collect information and update bypassing unit */
+    DataBypass::RegisterStage current_stage_of_communication = 0_RSG;
+
+    for ( auto& port:rps_stages_2_bypassing_unit)
```

Comment:
I still don't get the idea why you need feedback. Data bypass unit should know everything about instructions pushed into pipeline. If I am wrong, please share an example.